cpp/src/IceDiscovery/Makefile.mk

```diff
@@ -2,11 +2,11 @@
 
 $(project)_libraries := IceDiscovery
 
-# Tell the slice2cpp rule to ignore the top-level IceDiscovery include directory.
-IceDiscovery_includedir                 := $(project)/generated/IceDiscovery
-
+IceDiscovery_sliceflags                 := --include-dir IceDiscovery
```

Comment:
The header gets generated in the top-level include/generated/IceDiscovery directory.